### PR TITLE
[quant][fx][graphmode] Run symbolic_trace in quantization

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #45920 [quant][fx][graphmode][api] Change API for custom module
* **#45919 [quant][fx][graphmode] Run symbolic_trace in quantization**

Summary:
As discussed with JIT team, we'll run symbolic trace in quantization functions
prepare_fx now takes orginal pytorch model (torch.nn.Module) instead of `GraphModule` as input

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D24145857](https://our.internmc.facebook.com/intern/diff/D24145857)

### CODE REVIEW

**Review by z-a-f — APPROVED** [review]

File: torch/quantization/quantize_fx.py [hunk 1]
```diff
@@ -20,23 +23,36 @@ def fuse_fx(graph_module, inplace=False):
...
       is a submodule of the parent module that is not inlined in the
 forward graph of the parent module,
       the way we quantize standalone module is described in:
       :func:`~torch.quantization._prepare_standalone_module_fx`
     """
-    _check_is_graph_module(graph_module)
-    graph_module = fuse_fx(graph_module, inplace)
+    # symbolically trace the model
+    standalone_modules = qconfig_dict.get('standalone_module_name', [])
+    class CustomTracer(Tracer):
```
Comment (raghuramank100) [inline torch/quantization/quantize_fx.py:39]: Can we move the definition of this class out of the function and instantiate it with standalone_modules as an argument?
```python
class CustomTracer(Tracer):
   def __init__(self, standalone_modules):
      super(CustomTracer,self).__init__()
      self.standalone_modules = standalone_modules

   def is_leaf_module(self, m, module_qualified_name):
            return (m.__module__.startswith('torch.nn') and \
            not isinstance(m, torch.nn.Sequential)) or \
            module_qualified_name in self.standalone_modules

File: torch/quantization/quantize_fx.py [hunk 2]
```diff
@@ -51,14 +67,32 @@ def _prepare_standalone_module_fx(graph_module, qconfig_dict, inplace=False):
...
 
-def prepare_fx(graph_module, qconfig_dict, inplace=False):
+    Example:
+    ```python
+    from torch.quantization import fuse_fx
+    m = Model().eval()
+    m = fuse_fx(m)
+    ```
+    """
+    assert not model.training, 'fuse_fx only works on models in eval mode'
```
Comment (raghuramank100) [inline torch/quantization/quantize_fx.py:87]: Where is fusion done for QAT?

**Review by raghuramank100 — CHANGES_REQUESTED** [review]
Some suggestions, thanks!

File: torch/quantization/quantize_fx.py (hunk 2 above)
Comment (jerryzh168) [reply torch/quantization/quantize_fx.py:87]: this is the top level fusion function. we have another function `_fuse_fx` which does not check this flag. since fusion for QAT only makes sense for quantization so i didn't expose that

File: torch/quantization/quantize_fx.py (hunk 1 above)
Comment (jerryzh168) [reply torch/quantization/quantize_fx.py:39]: sure

**Review by raghuramank100 — APPROVED** [review]
